config-model/src/main/java/io/strimzi/kafka/config/model/ConfigModel.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.config.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+
+/**
+ * A model of a particular configuration parameter.
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ConfigModel {
```

Comment:
is it going to be a generic config model useable even for validating Kafka topic configuration for example?

Reply:
That should be possible, yes.